### PR TITLE
enh: Ticket Email - a better Text Email for confirmed ticket orders

### LINKED ISSUE #7148: Ticket Email: Create a better Text Email for confirmed ticket orders
We need a better ticket email. An HTML email would require a number of changes, but we can already create a better text email to improve things in the first step. An Email sample below.

 Questions: 
* What if the user bought several tickets, what link do we provide?
* Would it be possible to create a link string to provide the option "Add to my calendar", e.g. Google, Outlook, iCal

---

SUBJECT: Your Tickets for [event name]

Hello,

this is your order confirmation for [Event Name] ([event page link]) organized by [organization name].

YOUR TICKETS

Please download your tickets here: [link to ticket page]
You need to be logged in to access your ticket.

ORDER SUMMARY

Date: [date of order]
Order #: [order number]
Ordered by: [Name of Person who ordered]

Type: [Ticket Type]
Quantity: [Number of Tickets of Ticket Type bought]

*/ If the user buys different ticket types show additional ticket types and quantity beloe /*

Type: [Ticket Type]
Quantity: [Number of Tickets of Ticket Type bought]

ABOUT EVENT

Date: [Date of Event]
Time: [Time of Event with tiemzone]
Location: [Address of Event or link to virtual event]
OpenStreetMap: [Link to Location in OpenStreetMap (only show if it is an venue based event]

Add to my calendar:
Google: [link string]
Office 365: [link string]
Outlook: [link string]
iCal: [link string]
Yahoo: [link string]

YOUR ACCOUNT

Login to your account and manage your orders: https://eventyay.com/login

This email was sent to [Email address of recipient here]
Copyright © [Current Year] eventyay.com

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes #7148 

#### Changes proposed in this pull request:

Screenshot from 2020-08-13 18-33-54
- adding add to google calendar link
- adding event_map_link
- other required changes in Email template

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

**Comment by maze-runnar** [comment]
@iamareebjamal please take a look.

File: app/api/helpers/mail.py [hunk 1]
```diff
@@ -383,6 +385,18 @@ def send_email_to_attendees(order, purchaser_id, attachments=None):
                 html=MAILS[TICKET_PURCHASED_ATTENDEE]['message'].format(
                     my_tickets_url=frontend_url + '/my-tickets',
                     event_name=order.event.name,
+                    event_location = order.event.location_name,
+                    event_url = "https://eventyay.com/e/" + order.event.identifier,
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py:389]:
```suggestion
                    event_url = "https://eventyay.com/e/" + order.event.identifier,
```

Why eventyay?

File: app/api/helpers/mail.py [hunk 2]
```diff
@@ -374,6 +374,8 @@ def send_email_to_attendees(order, purchaser_id, attachments=None):
             )
         else:
             # The Ticket holder is not the purchaser
+            start_date = "".join(str(order.event.starts_at.date()).split("-"))
+            end_date = "".join(str(order.event.ends_at.date()).split("-"))
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py:378]: Use proper library methods to parse and convert date to event timezone

**Comment by iamareebjamal** [comment]
order.tickets will give you required information

File: app/api/helpers/mail.py (hunk 2 above)
Comment (maze-runnar) [reply app/api/helpers/mail.py:378]: ok

File: app/api/helpers/mail.py (hunk 1 above)
Comment (maze-runnar) [reply app/api/helpers/mail.py:389]: sorry, replacing it with frontend_url

**Comment by maze-runnar** [comment]
> order.tickets will give you the required information

order.tickets is returning  'InstrumentedList' as for one to many,  how to get an individual ticket_types, and names by 
 'InstrumentedList'

**Comment by iamareebjamal** [comment]
By iterating over them

**Comment by maze-runnar** [comment]
@iamareebjamal please review.
Screenshot from 2020-08-14 17-46-58

File: app/api/helpers/mail.py [hunk 3]
```diff
@@ -374,8 +374,8 @@ def send_email_to_attendees(order, purchaser_id, attachments=None):
             )
         else:
             # The Ticket holder is not the purchaser
-            start_date = "".join(str(order.event.starts_at.date()).split("-"))
-            end_date = "".join(str(order.event.ends_at.date()).split("-"))
+            start_date = (order.event.starts_at.date()).strftime('%Y%m%d')
+            end_date = (order.event.ends_at.date()).strftime('%Y%m%d')
```
Comment (iamareebjamal) [inline app/api/helpers/mail.py]: You need to convert the time to event timezone

Comment (maze-runnar) [reply app/api/helpers/mail.py]: done

**Comment by maze-runnar** [comment]
@iamareebjamal please take a look.

**Comment by iamareebjamal** [comment]
Good, but look at other places how the conversion is done. For now, let's move from these issues

**Comment by iamareebjamal** [comment]
Superseded by #7413

**Pull request closed without merging**